diff --git a/node_modules/@angular-devkit/build-angular/src/angular-cli-files/models/webpack-configs/common.js b/node_modules/@angular-devkit/build-angular/src/angular-cli-files/models/webpack-configs/common.js
index ad1abbc..1fc008d 100644
--- a/node_modules/@angular-devkit/build-angular/src/angular-cli-files/models/webpack-configs/common.js
+++ b/node_modules/@angular-devkit/build-angular/src/angular-cli-files/models/webpack-configs/common.js
@@ -19,6 +19,7 @@ const require_project_module_1 = require("../../utilities/require-project-module
 const utils_1 = require("./utils");
 const ProgressPlugin = require('webpack/lib/ProgressPlugin');
 const CircularDependencyPlugin = require('circular-dependency-plugin');
+const NormalModuleReplacementPlugin = require('webpack/lib/NormalModuleReplacementPlugin');
 const TerserPlugin = require('terser-webpack-plugin');
 const StatsPlugin = require('stats-webpack-plugin');
 // tslint:disable-next-line:no-any
@@ -36,7 +37,10 @@ function getCommonConfig(wco) {
         throw new Error('Cannot locate node_modules directory.');
     }
     // tslint:disable-next-line:no-any
-    const extraPlugins = [];
+    const extraPlugins = [
+        new NormalModuleReplacementPlugin(/knex/, 'node-noop'),
+        new NormalModuleReplacementPlugin(/redis/, 'node-noop'),
+    ];
     const entryPoints = {};
     if (buildOptions.main) {
         entryPoints['main'] = [path.resolve(root, buildOptions.main)];
@@ -226,7 +230,7 @@ function getCommonConfig(wco) {
             : 'development',
         devtool: false,
         resolve: {
-            extensions: ['.ts', '.tsx', '.mjs', '.js'],
+            extensions: ['@dps.js', '.ts', '.tsx', '.mjs', '.js'],
             symlinks: !buildOptions.preserveSymlinks,
             modules: [
                 wco.tsConfig.options.baseUrl || projectRoot,
