diff --git a/node_modules/enhanced-resolve/lib/AppendPlugin.js b/node_modules/enhanced-resolve/lib/AppendPlugin.js
index 7f488fc..4b9a691 100644
--- a/node_modules/enhanced-resolve/lib/AppendPlugin.js
+++ b/node_modules/enhanced-resolve/lib/AppendPlugin.js
@@ -16,15 +16,22 @@ module.exports = class AppendPlugin {
 		resolver
 			.getHook(this.source)
 			.tapAsync("AppendPlugin", (request, resolveContext, callback) => {
-				const obj = Object.assign({}, request, {
-					path: request.path + this.appending,
-					relativePath:
-						request.relativePath && request.relativePath + this.appending
-				});
+                let localAppending = this.appending;
+                if (this.appending === '@dps.js'
+                        && (request.path.indexOf('@dps\\mycms-frontend-commons') >= 0
+                                || request.path.indexOf('@dps/mycms-frontend-commons') >= 0)) {
+                        localAppending = '.js';
+                        // console.log("remapping @dps.js ts->js request.path", request.path, this.appending, localAppending);
+                    }
+                const obj = Object.assign({}, request, {
+                    path: request.path + localAppending,
+                    relativePath:
+                        request.relativePath && request.relativePath + localAppending
+                });
 				resolver.doResolve(
 					target,
 					obj,
-					this.appending,
+                    localAppending,
 					resolveContext,
 					callback
 				);
