diff --git a/node_modules/webpack/node_modules/schema-utils/src/validateOptions.js b/node_modules/webpack/node_modules/schema-utils/src/validateOptions.js
index 363af44..21f5440 100644
--- a/node_modules/webpack/node_modules/schema-utils/src/validateOptions.js
+++ b/node_modules/webpack/node_modules/schema-utils/src/validateOptions.js
@@ -27,8 +27,8 @@ const validateOptions = (schema, options, name) => {
     schema = fs.readFileSync(path.resolve(schema), 'utf8');
     schema = JSON.parse(schema);
   }
-
   if (!ajv.validate(schema, options)) {
+    console.error('webpack schema-validation error', ajv.errors, options, schema);
     throw new ValidationError(ajv.errors, name);
   }
 
