patch-package
--- a/node_modules/@angular/cli/models/webpack-configs/common.js
+++ b/node_modules/@angular/cli/models/webpack-configs/common.js
@@ -158,7 +158,7 @@ function getCommonConfig(wco) {
     catch (e) { }
     return {
         resolve: {
-            extensions: ['.ts', '.js'],
+            extensions: ['@dps.js', '.ts', '.js'],
             modules: ['node_modules', nodeModules],
             symlinks: !buildOptions.preserveSymlinks,
             alias
